internal/iam/resource.go

```diff
@@ -33,7 +33,7 @@ type Resource interface {
 	GetScope(ctx context.Context, r db.Reader) (*Scope, error)
 
 	// Type of Resource (Target, Policy, User, Group, etc)
-	ResourceType() resource.Type
+	GetResourceType() resource.Type
```

Comment:
Yeah I agree it would be nice to have the iam package use the boundary.Resource type. We haven't tackled the iam package yet so don't know if this will get in the way but I think this interface is a superset of `boundary.Resource`.